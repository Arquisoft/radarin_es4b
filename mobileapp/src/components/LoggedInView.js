import React, {useState, useEffect, useRef} from 'react';
import {View, AppState} from 'react-native';
import {Divider} from 'react-native-elements';
import LocationSwitch from './LocationSwitch.js';
import UserInfo from './UserInfo.js';
import DistanceSlider from './DistanceSlider.js';
import Map from './Map.js';
import {getObject, storeObject, getValue, storeValue} from '../storage.js';
import {
  startFriendUpdates,
  setMaxDistance,
  setForegroundFriendsHandler,
  getCurrentFriendsClose,
  setOnFriendNotificationCallback,
} from '../friends.js';
import * as CurrentUser from '../user.js';
import {setForegroundLocationHandler} from '../location.js';

const LoggedInView = ({user, changeUser}) => {
  const [mapRegion, setMapRegion] = useState({
    latitude: 32.5245988789034,
    longitude: -3.0120493099093437,
    latitudeDelta: 26.71393659121558,
    longitudeDelta: 13.803074732422829,
  });
  const [distance, setDistance] = useState(100);
  const [lastLocation, setLastlocation] = useState(null);
  const [friends, setFriends] = useState([]);
  const didMount = useRef(false);
  const map = useRef(null);

  useEffect(() => {
    let mounted = true;
    
    setForegroundLocationHandler(location => {
      if (mounted) {
        setLastlocation(location);
        getCurrentFriendsClose();
      }
    });

    setForegroundFriendsHandler(friends => {
      if (mounted) setFriends(friends);
    });

    setOnFriendNotificationCallback(friend => {
      if (map.current) {
        // Espera a que se realice el primer centrado sobre la posición
        // del propio usuario y después centra sobre el amigo
        setTimeout(
          () =>
            map.current.centerMap({
              coords: {
                latitude: friend.latitud,
                longitude: friend.longitud,
              },
            }),
          500,
        );
      }
    });

    getObject('region').then(region => {
      if (region !== null && mounted) setMapRegion(region);
    });

    getValue('maxDistance').then(value => {
      if (value !== null && mounted) setDistance(JSON.parse(value));
    });

    getObject(`${user.webId}-lastLocation`).then(location => {
      if (location !== null && mounted) {
        setLastlocation(location);
        CurrentUser.setLastUserLocation(location);
      }
    });

    startFriendUpdates();

    AppState.addEventListener('change', handleAppStateChange);

    return () => {
      mounted = false;
      AppState.removeEventListener('change', handleAppStateChange);
    };
  }, []);

  const handleAppStateChange = state => {
    if (state === 'active') {
      getCurrentFriendsClose();
      getObject(`${user.webId}-lastLocation`).then(location => {
        if (location !== null) setLastlocation(location);
      });
    }
  };

  useEffect(() => {
    let mounted = didMount.current;
    let onTimeout = false;
    if (didMount.current) {
      if (!onTimeout) {
        onTimeout = true;
        setTimeout(() => {
          if (mounted) {
            storeObject('region', mapRegion).then(() => (onTimeout = false));
          }
        }, 1000);
      }
    } else didMount.current = true;
    return function cleanup() {
      mounted = false;
    };
  }, [mapRegion]);

  useEffect(() => {
    let mounted = didMount.current;
    let onTimeout = false;
    if (didMount.current) {
      setMaxDistance(distance);
      if (!onTimeout) {
        onTimeout = true;
        setTimeout(() => {
          if (mounted) {
            getCurrentFriendsClose();
            storeValue('maxDistance', JSON.stringify(distance)).then(
              () => (onTimeout = false),
            );
          }
        }, 500);
      }
    } else didMount.current = true;
    return function cleanup() {
      mounted = false;
    };
  }, [distance]);

  useEffect(() => {
    if (didMount.current) {
      if (map.current) map.current.centerMap(lastLocation);
    }
  }, [lastLocation]);

  return (
    <View style={{flex: 1}}>
      <View
        style={{
          marginStart: 15,
          marginEnd: 15,
        }}>
        <UserInfo user={user} changeUser={changeUser} />
        <Divider />
        <LocationSwitch />
        <DistanceSlider distance={distance} changeDistance={setDistance} />
      </View>
      <Map
        ref={map}
        region={mapRegion}
        changeRegion={setMapRegion}
        lastUserLocation={lastLocation}
        locationDistance={distance}
        friends={friends}
      />
    </View>
  );
};

export default LoggedInView;
