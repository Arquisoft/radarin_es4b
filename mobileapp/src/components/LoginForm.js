import React, {useState} from 'react';
import {StyleSheet, View, Text, TextInput, Linking} from 'react-native';
import {Icon} from 'react-native-elements';
import Toast from 'react-native-simple-toast';
import {authenticate} from '../api/api.js';
import getText from '../i18n.js';
import CustomLoadingButton from './CustomLoadingButton';

const LoginForm = ({changeUser}) => {
  const [idp, setIdp] = useState();
  const [username, setUsername] = useState();
  const [password, setPassword] = useState();

  const authenticateWithCredentials = async credentials => {
    let user = await authenticate(credentials);
    user ? changeUser(user) : Toast.show(getText('toastLogIn'));
  };

  return (
    <View>
      <View style={styles.container}>
        <Text style={styles.title}>{getText('logIn')}</Text>
        <Text style={styles.label}>{getText('provider')}</Text>
        <TextInput
          style={styles.input}
          onChangeText={text => setIdp(text)}
          placeholder={getText('providerPlaceholder')}></TextInput>
        <Text style={styles.label}>{getText('username')}</Text>
        <TextInput
          style={styles.input}
          onChangeText={text => setUsername(text)}
          placeholder={getText('usernamePlaceholder')}></TextInput>
        <Text style={styles.label}>{getText('password')}</Text>
        <TextInput
          style={styles.input}
          secureTextEntry={true}
          onChangeText={text => setPassword(text)}
          placeholder={getText('passwordPlaceholder')}></TextInput>
        <CustomLoadingButton
          action={() => authenticateWithCredentials({idp, username, password})}
          text={getText('enviar')}
          customContainerStyles={styles.button}
        />
      </View>
      <View style={styles.infoContainer}>
        <Icon name={'info'} />
        <Text style={styles.infoText}>{getText('loginInfo')}</Text>
        <Text style={styles.infoText}>
          {getText('getAPod')}
          <Text
            style={styles.infoLink}
            onPress={() =>
              Linking.openURL('https://solidproject.org/users/get-a-pod')
            }>
            {getText('here')}
          </Text>
          .
        </Text>
        <Text style={styles.infoText}>
          {getText('trustedApps')}
          <Text
            style={styles.infoLink}
            onPress={() =>
              Linking.openURL(
                'https://github.com/solid/userguide#manage-your-trusted-applications',
              )
            }>
            {getText('here')}
          </Text>
          .
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginStart: 'auto',
    marginEnd: 'auto',
  },
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    marginBottom: 10,
    alignSelf: 'flex-start',
  },
  label: {
    fontSize: 16,
    marginBottom: 10,
    alignSelf: 'flex-start',
  },
  input: {
    height: 40,
    width: 270,
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 4,
    marginBottom: 20,
  },
  button: {
    alignSelf: 'flex-end',
  },
  infoContainer: {
    margin: 25,
    marginTop: 30,
    padding: 15,
    backgroundColor: '#e3e3e3',
    borderRadius: 4,
    alignItems: 'flex-start',
  },
  infoText: {
    marginTop: 7,
  },
  infoLink: {
    color: '#3d48e3',
    textDecorationLine: 'underline',
  },
});

export default LoginForm;
