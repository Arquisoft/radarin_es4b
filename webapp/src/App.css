.App {
  text-align: center;
}

.App-logo {
  height: 20vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 3em;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.App-content {
  padding-top:5vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center; 
}

.Friends {
  display: flex;
  flex-direction: row;
}

.UsersList {
  width: 30%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
}

body {
  overflow-x: hidden;
}


@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}