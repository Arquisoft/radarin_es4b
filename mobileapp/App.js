import React, {useEffect, useState, useRef} from 'react';
import {ScrollView, View} from 'react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Divider} from 'react-native-elements';
import Header from './src/components/Header';
import LocationSwitch from './src/components/LocationSwitch';
import LoginForm from './src/components/LoginForm';
import UserInfo from './src/components/UserInfo';
import DistanceSlider from './src/components/DistanceSlider';
import {getObject, storeObject} from './src/storage.js';

const App = () => {
  const [user, setUser] = useState();
  const didMount = useRef(false);

  useEffect(() => {
    getObject('user').then(user => {
      if (user) setUser(user);
    });
  }, []);

  useEffect(() => {
    if (didMount.current) storeObject('user', user);
    else didMount.current = true;
  }, [user]);

  return (
    <SafeAreaProvider>
      <ScrollView>
        <Header></Header>
        <View
          style={{
            marginTop: 30,
            flex: 1,
          }}>
          {!user && <LoginForm changeUser={setUser}></LoginForm>}
          {user && (
            <View
              style={{
                marginStart: 15,
                marginEnd: 15,
              }}>
              <UserInfo user={user} changeUser={setUser} />
              <Divider />
              <LocationSwitch webId={user.webId} />
              <DistanceSlider />
            </View>
          )}
        </View>
      </ScrollView>
    </SafeAreaProvider>
  );
};

export default App;
