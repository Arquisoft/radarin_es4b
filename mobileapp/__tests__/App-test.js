/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import getText from '../src/i18n.js';
import {currentlLocale} from 'react-native-i18n';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import LoginForm from '../src/components/LoginForm';
import LoggedInView from '../src/components/LoggedInView';

test('app renders correctly', () => {
  renderer.create(<App />);
});

test('applies the expected language', () => {
  if (currentlLocale() === 'en')
    expect(getText('location')).toBe('Send my location');
  else if (currentlLocale() === 'es')
    expect(getText('location')).toBe('Enviar mi ubicación');
});

test('log in form renders correctly', () => {
  const changeUser = user => {
    console.log(user);
  };

  const logInForm = renderer
    .create(<LoginForm changeUser={changeUser}></LoginForm>)
    .toJSON();
  expect(logInForm).toMatchSnapshot();
});

test('user with valid credentials can log in', async done => {
  const expectUser = {
    webId: 'https://davidaf.solidcommunity.net/profile/card#me',
    name: 'David Álvarez',
    photo: 'https://davidaf.solidcommunity.net/profile/fotosolid.png',
  };

  fetch.mockResponseOnce(JSON.stringify(expectUser), {status: 200});

  const changeUser = user => {
    expect(JSON.stringify(user)).toBe(JSON.stringify(expectUser));
    done();
  };

  const {getByPlaceholderText, getByText} = render(
    <LoginForm changeUser={changeUser}></LoginForm>,
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('providerPlaceholder')),
    'https://solidcommunity.com',
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('usernamePlaceholder')),
    'david',
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('passwordPlaceholder')),
    '1234',
  );

  await act(async () => {
    fireEvent.press(getByText(getText('enviar')));
  });
});

test('user with invalid credentials cant log in', async () => {
  fetch.mockResponseOnce('Error', {status: 403});

  const currentUser = null;

  const changeUser = user => {
    fail('it should not reach here');
    currentUser = user;
  };

  const {getByPlaceholderText, getByText} = render(
    <LoginForm changeUser={changeUser}></LoginForm>,
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('providerPlaceholder')),
    'https://provider.com',
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('usernamePlaceholder')),
    'Pepito',
  );
  fireEvent.changeText(
    getByPlaceholderText(getText('passwordPlaceholder')),
    '7777',
  );
  await act(async () => {
    fireEvent.press(getByText(getText('enviar')));
  });

  // Comprueba que el usuario no se ha cambiado
  expect(currentUser).toBe(null);
});

test('logged in view renders correctly', () => {
  const currentUser = {
    webId: 'https://davidaf.solidcommunity.net/profile/card#me',
    name: 'David Álvarez',
    photo: 'https://davidaf.solidcommunity.net/profile/fotosolid.png',
  };

  const changeUser = user => {
    fail('it should not reach here');
    currentUser = user;
  };

  const loggedInView = renderer
    .create(
      <LoggedInView user={currentUser} changeUser={changeUser}></LoggedInView>,
    )
    .toJSON();
  expect(loggedInView).toMatchSnapshot();
});

test('user can log out', async done => {
  const currentUser = {
    webId: 'https://davidaf.solidcommunity.net/profile/card#me',
    name: 'David Álvarez',
    photo: 'https://davidaf.solidcommunity.net/profile/fotosolid.png',
  };

  const changeUser = user => {
    expect(user).toBe(null);
    done();
  };

  const {getByText} = render(
    <LoggedInView user={currentUser} changeUser={changeUser}></LoggedInView>,
  );

  await act(async () => {
    fireEvent.press(getByText(getText('logOut')));
  });
});
