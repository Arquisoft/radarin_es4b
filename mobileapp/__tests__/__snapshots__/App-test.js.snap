// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log in form renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginEnd": "auto",
        "marginStart": "auto",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Log in
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Solid identity provider:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="e. g. https://solidcommunity.net"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "width": 270,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Username:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "width": 270,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Password:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "width": 270,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#007bff",
            "borderRadius": 4,
            "color": "white",
            "fontSize": 16,
            "height": 40,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 100,
          }
        }
      >
        Send
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#e3e3e3",
        "borderRadius": 4,
        "margin": 25,
        "marginTop": 30,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#242424",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "marginTop": 7,
        }
      }
    >
      To use Radarin you need to have a Solid POD and have added https://solid-node-client as a trusted app
    </Text>
    <Text
      style={
        Object {
          "marginTop": 7,
        }
      }
    >
      If you don't have a POD, you can get one 
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#3d48e3",
            "textDecorationLine": "underline",
          }
        }
      >
        here
      </Text>
      .
    </Text>
    <Text
      style={
        Object {
          "marginTop": 7,
        }
      }
    >
      If you don't know how to add a trusted app, you can find more information 
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#3d48e3",
            "textDecorationLine": "underline",
          }
        }
      >
        here
      </Text>
      .
    </Text>
  </View>
</View>
`;

exports[`logged in view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginEnd": 15,
        "marginStart": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "marginBottom": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "height": 50,
            "opacity": 1,
            "width": 50,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 25,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={25}
            onLoad={[Function]}
            source={
              Object {
                "uri": "https://davidaf.solidcommunity.net/profile/fotosolid.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            transition={true}
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "fontSize": 16,
            "marginEnd": "auto",
            "marginStart": 10,
          }
        }
      >
        David Álvarez
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#007bff",
              "borderRadius": 4,
              "color": "white",
              "fontSize": 16,
              "height": 40,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 100,
            }
          }
        >
          Log out
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#bcbbc1",
          "height": 0.5,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#242424",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginEnd": "auto",
            "marginStart": 7,
          }
        }
      >
        Send my location
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#9ec2ff"
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="#f4f3f4"
        tintColor="#d6d6d6"
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#242424",
                      "fontSize": 20,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 3,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginStart": 7,
              }
            }
          >
            Max distance for friend notifications:
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginEnd": 10,
                "textAlign": "right",
              }
            }
          >
            100
             m
          </Text>
        </View>
      </View>
      <View
        allowTouchTrack={true}
        animationType="timing"
        onLayout={[Function]}
        onValueChange={[Function]}
        orientation="horizontal"
        step={10}
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginTop": 5,
            "width": 250,
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        thumbTouchSize={
          Object {
            "height": 23,
            "width": 23,
          }
        }
        value={100}
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "#b3b3b3",
              "borderRadius": 2,
              "height": 3,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#3f3f3f",
              "borderRadius": 2,
              "height": 0,
              "marginTop": -0,
              "position": "absolute",
              "width": 0,
            }
          }
        />
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "#007bff",
              "borderRadius": 20,
              "height": 22,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 22,
            }
          }
          testID="sliderThumb"
        />
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <AIRMap
        handlePanDrag={false}
        initialRegion={
          Object {
            "latitude": 32.5245988789034,
            "latitudeDelta": 26.71393659121558,
            "longitude": -3.0120493099093437,
            "longitudeDelta": 13.803074732422829,
          }
        }
        moveOnMarkerPress={false}
        onChange={[Function]}
        onLayout={[Function]}
        onMapReady={[Function]}
        onMarkerPress={[Function]}
        onRegionChange={[Function]}
        region={null}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
